stixcore/processing/levelb.py

```diff
@@ -0,0 +1,150 @@
+import os
+import logging
+from time import perf_counter
+from pathlib import Path
+
+import numpy as np
+
+from astropy.table import unique, vstack
+from astropy.table.table import QTable, Table
+
+from stixcore.io.fits.processors import SEC_IN_DAY, FitsLBProcessor
+from stixcore.io.soc.manager import SOCManager
+from stixcore.tmtc.packets import TMTC
+from stixcore.util.logging import get_logger
+
+__all__ = ['LevelB']
+
+
+logger = get_logger(__name__, level=logging.DEBUG)
+
+
+class LevelB:
+    """
+    Class representing level binary data.
+    """
+    def __init__(self, control, data):
+        self.level = 'LB'
+        self.type = f"{control['service_type'][0]}-{control['service_subtype'][0]}"
+        self.control = control
+        self.data = data
+
+        synced_times = (self.control['scet_coarse'] >> 31) != 1
+        self.control['scet_coarse'] = np.where(synced_times, self.control['scet_coarse'],
+                                               self.control['scet_coarse'] ^ 2**31)
+        self.control['time_sync'] = synced_times
+        scet = self.control['scet_coarse'] + self.control['scet_coarse']/2**16
```

Comment:
Can probably replace some of this logic with our DateTime class but this can be done in a separate issue commenting here before I forget

Reply:
open for now